simexpal/build.py

```diff
@@ -264,12 +264,29 @@ def git_ref_changed():
 						build.clone_dir,
 						generic_tag])
 			else:
-				# Recreate the source directory
-				util.try_rmtree(build.source_dir)
-				util.try_mkdir(build.source_dir)
-
-				# Clone the git repository into the build.source_dir
-				subprocess.check_call(['git', 'clone', build.info.git_repo, build.source_dir])
+				branch = build.revision.version_for_build(build.name)
+				if os.path.isdir(build.source_dir) and not reclone:
+					# Repository already cloned, check for local changes
+					status = subprocess.check_output(['git', 'status', '--porcelain'], cwd=build.source_dir)
```

Comment:
But now you're trying to check out to the branch given as the value for the branch, never mind if you're already on the branch? If so, this should be changed. If you're already on the branch then all these checks don't have to be proceeded with.